ledger-core/virtual/object/object.go

```diff
@@ -398,14 +404,74 @@ func (sm *SMObject) createWaitPendingOrderedSM(ctx smachine.ExecutionContext) {
 	}
 }
 
-func (sm *SMObject) migrate(ctx smachine.MigrationContext) smachine.StateUpdate {
+func (sm *SMObject) stepMigration(ctx smachine.MigrationContext) smachine.StateUpdate {
 	if sm.GetState() == Unknown {
 		ctx.Log().Trace("SMObject migration happened when object is not ready yet")
 		return ctx.Stop()
 	}
 
 	ctx.UnpublishAll()
 
+	// store migration pulse
+	sm.MigratePulseNumber = sm.pulseSlot.PulseData().PulseNumber
```

Comment:
you don't need to store it in object. You always can call sm.pulseSlot.PulseData().PulseNumber. It will not change

Reply:
Fixed